src/components/UiPhoto/FileField.vue

```diff
@@ -0,0 +1,463 @@
+      v-if="errorMessage"
+    >
+      {{ errorMessage }}
+    </div>
+  </div>
+</template>
+
+<script>
+import { FileUtil, FileNotPresentInEventError } from '@/utils/file.util'
+import { Document } from '@tokend/js-sdk'
+
+const MAX_FILE_MEGABYTES = 32
+const FILE_EXTENSIONS = ['jpg', 'png', 'jpeg', 'pdf']
+
+export default {
+  props: {
+    value: { type: Document, default: null },
+    labelMain: { type: String, default: 'Logo (Optional)' },
+    labelTitle: { type: String, default: 'Drag a file here or click to browse' },
+    labelFileName: { type: String, default: 'Selected file: ' },
+    labelDraged: { type: String, default: 'Drag another file or click to browse' },
+    documentType: { type: String, default: 'default' },
+    fileExtensions: { type: Array, default: _ => FILE_EXTENSIONS },
+    maxSize: { type: Number, default: MAX_FILE_MEGABYTES },
+    note: { type: String, default: 'image (Max 32mb)' },
+    errorMessage: { type: String, default: undefined },
+    minWidth: { type: Number, default: 0 },
+    minHeight: { type: Number, default: 0 },
+  },
+
+  data: _ => ({
+    document: null,
+    isFileDragged: false,
+    documentUrl: '',
+  }),
+
+  computed: {
+    maxSizeBytes () {
+      return this.maxSize * 1024 * 1024
+    },
+
+    isImageSelected () {
+      const mimeType = this.document.mimeType
+      return mimeType && mimeType.includes('image')
+    },
+
+    acceptedExtensions () {
+      return this.fileExtensions
+        .map(item => `.${item.toUpperCase()}`)
+        .join(', ')
+    },
+  },
+  watch: {
+    'value': {
+      handler (value) {
+        this.document = value
+        if (value) this.loadDocumentUrl(value)
+      },
+      immediate: true
+    }
+  },
+  methods: {
+    async loadDocumentUrl (document) {
+      if (document.key) {
+        this.documentUrl = await document.getPrivateUrl()
+      } else {
+        this.documentUrl = await FileUtil.getDataUrl(document.file)
+      }
+    },
+    async onChange (event) {
+      try {
+        const file = FileUtil.getFileFromEvent(event)
+        if (await this.validateFile(file)) {
+          this.documentUrl = await FileUtil.getDataUrl(file)
+          this.document = Document({
+            mimeType: file.type,
+            type: this.documentType,
+            name: file.name,
+            file: file,
+          })
+          this.$emit('input', this.document)
+        } else {
+          this.$emit('error')
+        }
+      } catch (e) {
+        if (e instanceof FileNotPresentInEventError) {
+          this.reset()
+        } else {
+          console.error(e)
+          this.$emit('error')
+        }
+        return false
+      }
+    },
+
+    reset () {
+      this.$el.querySelector('input').value = ''
+      this.document = null
+      this.documentUrl = ''
+      this.$emit('input', this.document)
+    },
+
+    async validateFile (file) {
+      if (!this.isValidFileType(file)) {
+        this.$emit('error')
```

Comment:
let's provide a prop with a handler for each. type of error. should be possible to do smth like:

```
<ui-photo
 @err-invalid-file-size="someHandler"
 @err-invalid-file-type="someAnotherHandler"
 @err-invalid-file-dimensions="oneMoreHandler"
/>
```